### ISSUE TITLE
Troubles with Restify

### ISSUE THREAD

**Issue by PhilippCh** [DylanPiercey/auto-sni#8]
I have some issues getting auto-sni to work with Restify. I've followed your example, but was only able to make it work by changing your index.js Line 103 from

    // Let request through to provided https-server.
    httpsServer.emit('request', req, res)

to 

    // Let request through to provided https-server.
    httpsServer.server.emit('request', req, res)

After this I can get Restify to work using forceSSL = false. As soon as I change it to true though, all my requests get dropped in nirvana and the server responds with an unexpectedly closed connection. No amount of debugging could get me the exact point at where my connections are dropped. I can't even reach the SNICallback or handleChallenge. I can see you're passing SNICallback to createServer, but I can't seem to figure out how Restify (or node's https for that matter) is using it.

It'd be really great if you could help me out!

**Comment by DylanPiercey** [comment]
Hey @PhilippCh, thanks for creating the issue. I have limited experience with Restify so your help world be great to resolve this.

Could you try changing adding this below line 50?

```js
  // Special case for restify servers.
  if (httpsServer.server) httpsServer = httpsServer.server;
```

Let me know if this works and I'll add the fix, if not please shoot me a message on the gitter if you can.

**Comment by DylanPiercey** [comment]
@PhilippCh I added this fix to 0.9.12, I'm going to close this issue but let me know if this doesn't work for you.

**Comment by PhilippCh** [comment]
Thanks a lot for the fix! It's indeed working just fine now. Always better to have this in the official package than manually editing it :)

**Comment by DylanPiercey** [comment]
@PhilippCh glad to here it's working for you.

**Comment by bstahlhood** [comment]
This seems to still be a bug in 0.9.13. Following your example with Restify, I get a:
"TypeError: https.createServer is not a function"

**Comment by DylanPiercey** [comment]
@bstahlhood could you confirm that this is fixed in 0.10?

**Comment by bstahlhood** [comment]
@DylanPiercey Yes, it's fixed. Just having a different issue. I will try to debug with you on Gitter
